test/dataset/test_rolling.py

```diff
@@ -0,0 +1,306 @@
+    elif ds_name == "varying":
+        to_compare = generate_expected_dataset_varying_standard(pl)
+
+    assert to_compare, "dataset to compare with is not implemented"
+
+    return to_compare
+
+
+def generate_dataset(name):
+    dataset = None
+    if name == "constant":
+        _, _, dataset = constant_dataset()
+    elif name == "varying":
+        # Tests edge cases
+        # t0: start time of target
+        # ts: start time of desired range
+        # te: end time of desired range
+        # t1: end time of target
+        # ts < te, t0 <= t1
+        #
+        # start time index of rolling window is 20
+        # end time index of rolling window is 24
+        # ts = 2000-01-01 20:00:00
+        # te = 2000-01-02 00:00:00
+
+        ds_list = [
+            {  # test 1: ends after end time, te > t1
+                "target": [0.0 for i in range(30)],
+                "start": pd.Timestamp(2000, 1, 1, 0, 0),
+            },
+            {  # test 2: ends at the end time, te == t1
+                "target": [0.0 for i in range(25)],
+                "start": pd.Timestamp(2000, 1, 1, 0, 0),
+            },
+            {  # test 3: ends between start and end times, ts < t1 < te
+                "target": [0.0 for i in range(23)],
+                "start": pd.Timestamp(2000, 1, 1, 0, 0),
+            },
+            {  # test 4: ends on start time, ts == t1
+                "target": [0.0 for i in range(20)],
+                "start": pd.Timestamp(2000, 1, 1, 0, 0),
+            },
+            {  # test 5: ends before start time, t1 < ts
+                "target": [0.0 for i in range(15)],
+                "start": pd.Timestamp(2000, 1, 1, 0, 0),
+            },
+            {  # test 6: starts on start ends after end, ts == t0, te > t1
+                "target": [0.0 for i in range(10)],
+                "start": pd.Timestamp(2000, 1, 1, 20, 0),
+            },
+            {  # test 7: starts in between ts and te, ts < t0 < te < t1
+                "target": [0.0 for i in range(10)],
+                "start": pd.Timestamp(2000, 1, 1, 22, 0),
+            },
+            {  # test 8: starts on end time, te == t0
+                "target": [0.0 for i in range(10)],
+                "start": pd.Timestamp(2000, 1, 2, 0, 0),
+            },
+            {  # test 9: starts after end time, te < t0
+                "target": [0.0 for i in range(10)],
+                "start": pd.Timestamp(2000, 1, 2, 1, 0),
+            },
+            {  # test 10: starts after ts & ends before te, ts < t0 < t1 < te
+                "target": [0.0 for i in range(3)],
+                "start": pd.Timestamp(2000, 1, 1, 21, 0),
+            },
+        ]
+        dataset = ListDataset(ds_list, "H")
+    else:
+        raise ValueError
+    return dataset
+
+
+@pytest.mark.parametrize(
+    "prediction_length, unique",
+    [(p, u) for p in [-1, 0] for u in [True, False]],
+)
+def test_fails(prediction_length, unique):
```

Comment:
I think that's a bad test-name. When you want to select tests with `-k` you generally want to use more descriptive names.